src/algorithms/tracking/TrackParamTruthInit.cc

```diff
@@ -92,7 +93,7 @@ eicrecon::TrackParamTruthInit::produce(const edm4hep::MCParticleCollection* mcpa
         // Insert into edm4eic::TrackParameters, which uses numerical values in its specified units
         auto track_parameter = track_parameters->create();
         track_parameter.setType(-1); // type --> seed(-1)
-        track_parameter.setLoc({0.0, 0.0}); // 2d location on surface [mm]
+        track_parameter.setLoc({std::hypot(v.x, v.y), v.z}); // 2d location on surface [mm]
```

Comment:
Not sure why MCParticle has Vector3d for vertices, but whatever:
```suggestion
        track_parameter.setLoc({static_cast<float>(std::hypot(v.x, v.y)), static_cast<float>(v.z)}); // 2d location on surface [mm]
```
A choice between two casts and a double hypot or three casts and a float hypot... probably doesn't matter.